### DISCUSSION TITLE
User management in code server

### DISCUSSION THREAD

**Discussion by mohammedyusha** [coder/code-server#5591] [Q&A]
Hi,

Is this possible in code server?

a)	Code Server should create a folder like userid and change the directory to the created userid
b)	Should not allow to go back by using any means of terminal using command like (cd ..)

**Comment by jsjoeio** [comment]
Hmm...you'd probably have to implement your own thing in code-server to have such functionality. [coder/coder](https://github.com/coder/coder) is meant for teams and each user could have their own environment so maybe that's worth exploring? it's also open source.

**Reply by mohammedyusha** [reply]
Any suggestions for implementation?

**Reply by jsjoeio** [reply]
For implementing in code-server? Hmm...let's think out loud for a second. 

There is one instance of code-server running:
1. implement an auth layer in front of it like [Pomerium](https://www.pomerium.com/docs/guides/code-server) and allow specific users to login via GitHub
2. implement some type of middleware that runs a script to create folder like userid and change to that directory
3. patch code-server and prevent users from accessing other folders

I think 3 would be the hardest part. I don't know enough about file system privileges, etc. to implement something like that.

The easier thing to do (less work for you) would be to deploy coder/coder which is **built to solve this problem for you.** It's OSS so you could use it for free and then pay for licenses if you want Enterprise features.

**Reply by mohammedyusha** [reply]
Thanks for the suggestion, I will look into it.

**Reply by mohammedyusha** [reply]
Can coder be deployed on AWS EC2?, if yes is there a guide?

**Reply by lg-nineoneninethreenine** [reply]
Yeah. Just install Coder through the prebuilt binaries, and make sure Postgres is installed. Then just start the server, make sure you have an access URL, and just go start coding! Create your first user via the CLI, that's really important. I bet more info can be found at https://coder.com/docs and if you need more help, hop onto coder/coder and start a discussion!

P.S. if you can pay for a domain, you can use it on Coder. I can't find a guide now, but I know you can do that.

**Reply by lg-nineoneninethreenine** [reply]
Oh, the link is [this](https://coder.com/docs/coder-oss/latest).

**Comment by geiseri** [comment]
I am using trafik and authelia right now and mount the workspace directory with samba.  What I am really wanting though I think is somewhere closer to vscode dev containers that are create on the fly.  The user then authenticates, selects the devenv, performs the code checkout, starts the container and then trafik publishes it.  I think in most cases people don't really want multi-user, but more multi-workspace.   In a way you get multi-user for free in that aspect since you can bind a workspace to a user.

**Reply by jsjoeio** [reply]
That's super cool! This is very similar to what you can do with coder/coder too.

**Reply by geiseri** [reply]
Yeah, I mean coder is really overkill for what I need.  I have tried a few times to get it working, but its still pretty raw.  To be honest in the end getting coder to work with a custom template to digest a devenv workspace will be less work for me ;)
